### DISCUSSION TITLE
Maybe a way to some kind of same driver settings for all :ST7796-ILI9488-ILI9486-ILI9341... ? with ESP32

### DISCUSSION THREAD

**Discussion by TangerineDreamer** [Bodmer/TFT_eSPI#2239] [Compatible displays and setup files]
Hi,
Can someone explain me how it is possible:

I use ESP32 Dev Kit , here is my config:

----------------------------------------------------------------
//#define ILI9341_DRIVER 

//#define ILI9488_DRIVER
    #define ILI9486_DRIVER 

//#define ST7796_DRIVER 

//TFT Pins (HSPI)
#define TFT_MISO 12
#define TFT_MOSI 13
#define TFT_SCLK 14
#define TFT_CS   15  // Chip select control pin
#define TFT_DC    2  // Data Command control pin
#define TFT_RST   4  // Reset pin (could connect to RST pin)
//#define TFT_RST  -1  // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST

//Touch Pins (HSPI)
#define TOUCH_DO 12 
#define TOUCH_DIN 13 
#define TOUCH_CLK 14
#define TOUCH_CS 33//27 // Chip select pin (T_CS) of touch screen

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
//#define LOAD_FONT8N // Font 8. Alternative to Font 8 above, slightly narrower, so 3 digits fit a 160 pixel TFT
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT
#define SPI_FREQUENCY  20000000
#define SPI_READ_FREQUENCY  20000000
#define SPI_TOUCH_FREQUENCY  2500000

#define USE_HSPI_PORT

-------------------------------------------------------------------------------
And here's my sketch settings:

--------------------------------------------------------------------------------
#include "SPI.h"
SPIClass spiTFT(HSPI);

#include <TFT_eSPI.h> // Hardware-specific library
#include <SPI.h>

TFT_eSPI tft = TFT_eSPI();       // Invoke custom library

//and so on...

void setup()
{
Serial.begin(115200);
 
  spiTFT.begin(14,12,13,15);

//and son on...

void loop()
{}
-------------------------------------------------------------------------------

Ok,what i discover it that all the TFT's show the sketch,eg:meters
but with differences in colors between ILI9341 and ILI9488,ILI9486,ST7796
But if i choose in the tft_setup ILI9486 driver,all the TFT show the same colors in the "meters" sketch.

I can imagine a colors wrapper to get the right ones for all !!
I don't have the other TFT's drivers to test,maybe you can verify if i'm not dreaming :)

**Comment by TangerineDreamer** [comment]
I think it's the HSPI class that use this settings:

//TFT Pins (HSPI)
#define TFT_MISO 12
#define TFT_MOSI 13
#define TFT_SCLK 14
#define TFT_CS 15 // Chip select control pin
#define TFT_DC 2 // Data Command control pin
#define TFT_RST 4 // Reset pin (could connect to RST pin)
//#define TFT_RST -1 // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST

That makes the difference with the standart SPI settings,and the use of :
#define USE_HSPI_PORT in the setup,
So if you don't know whether your TFT works or not and don't see anything on the screen,it could be a solution
to see something.
I limited my investigations with ESP32 and to the TFT's i've got:
ST7796-ILI9488-ILI9486-ILI9341.
TFT 4.0, TFT 3.5, TFT 3.2, TFT 2.8

**Comment by Bodmer** [comment]
I think this is just luck. There seems to be limited standardisation for the register map and settings.

**Comment by TangerineDreamer** [comment]
Well it's by chance because i forgot to change the tft driver setup, while i was changing tft model and i had a big surprise
for christmas :)
what do you mean by standardisation?from the manufacturer?
